src/charm.py

```diff
@@ -169,32 +242,79 @@ def _on_collect_status(self, event: CollectStatusEvent):
         if not self._is_vault_service_started():
             event.add_status(WaitingStatus("Waiting for Vault service to start"))
             return
+        if not self._api_address:
+            event.add_status(WaitingStatus("No address received from Juju yet"))
+            return
+        if not self.tls.tls_file_available_in_charm(File.CA):
+            event.add_status(WaitingStatus("Certificate is unavailable in the charm"))
+            return
```

Comment:
Shouldn't those 2 validations go before the `self._is_vault_service_started()` validation?

Reply:
Yeah that probably makes sense. Admittedly, I was only considering the context of my changes.

Reply:
Yes. This may actually cause a crash as you identified above.